models/classes/LtiProvider/LtiProviderService.php

```diff
@@ -45,12 +50,23 @@ static function ($count, LtiProviderRepositoryInterface $implementation) {
         );
     }
 
+    public function searchByToolClientId(string $clientId): LtiProvider
+    {
+        foreach ($this->findAll() as $provider) {
+            if ($provider instanceof LtiProvider && $clientId === $provider->getToolClientId()) {
```

Comment:
```suggestion
            if ($clientId === $provider->getToolClientId()) {
```
Why check the instance?

Reply:
For some reason few of unit tests in `LtiProviderServiceTest`, expected `$provider` to be string.

Reply:
Those tests are wrong then, the method findAll only return LtiProvider objects. Keep this verification will cause misunderstanding in the code and are not necessary.